dataikuapi/dss/future.py

```diff
@@ -72,6 +72,8 @@ def wait_for_result(self):
         """
         Wait and get the future result
         """
+        if self.state.get('hasResult', False) and 'result' in self.state:
```

Comment:
I don't think we should have this additional `'result' in self.state` check:
- The source of truth telling whether or not the result is available is `hasResult`
- The presence of `result` is not guaranteed (even if the future is completed) because our JSON serializer drops null-valued properties (for instance `{"result": null}`  gets serialized as `{}` by DSS)

Reply:
Yes, if `result not in self.state` then this is handled by `self.state.get('result', None)` and the default value will be `None`
👍